### PR TITLE
fix: change return timestamp to return last outcome timestamp

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: 
- https://oat-sa.atlassian.net/browse/INF-230 
- https://oat-sa.atlassian.net/browse/INF-225
- https://oat-sa.atlassian.net/browse/INF-219
- https://oat-sa.atlassian.net/browse/INF-218

## What's Changed

- Return last outcome variable timestamp on AGS call, use all outcome declarations to check if test is fully graded

## How to test
- Create a test with item which has externally graded score, deliver and take a score.
- Call taoResultServer/api/DeliveryExecutionResults to update score and you will get last outcome variable timestamp
- Call again without updating the score and you will also receive last outcome variable timestamp
## Video

https://github.com/oat-sa/extension-tao-outcome/assets/118974926/b10ce132-9bf7-4259-b573-aa45a9420186

### CODE REVIEW

**Review by bartlomiejmarszal — COMMENTED** [review]
**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: test/Unit/models/Import/Service/SendCalculatedResultServiceTest.php [hunk 1]
```diff
@@ -309,7 +309,7 @@ private function createVariable(int $number, bool $isExternallyGraded): OutcomeV
             'identifier' => sprintf('OUTCOME_%d', $number),
             'cardinality' => 'single',
             'baseType' => 'float',
-            'epoch' => (string)time(),
+            'epoch' => '0.69485100 1690808122',
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply test/Unit/models/Import/Service/SendCalculatedResultServiceTest.php:312]: Hello Alberto :) Its a way to store timestamp with microseconds

File: models/Events/DeliveryExecutionResultsRecalculated.php [hunk 2]
```diff
@@ -32,14 +32,14 @@ class DeliveryExecutionResultsRecalculated implements Event
...
-    private int $timestamp;
+    private ?int $timestamp;
 
     public function __construct(
         DeliveryExecutionInterface $deliveryExecution,
         ?float $totalScore,
         ?float $totalMaxScore,
         bool $isFullyGraded,
-        int $timestamp
+        ?int $timestamp
```
Comment (pnal) [inline models/Events/DeliveryExecutionResultsRecalculated.php:42]: Is there any case for it to be null?

**Review by pnal — APPROVED** [review]
Good job!

File: models/Events/DeliveryExecutionResultsRecalculated.php (hunk 2 above)
Comment (Karol-Stelmaczonek) [reply models/Events/DeliveryExecutionResultsRecalculated.php:42]: Yes, getFinishTime() can return null 
image

Comment (pnal) [reply models/Events/DeliveryExecutionResultsRecalculated.php:42]: but we do not trigger this event if the execution not finished, by it is obvious from it's name DeliveryExecutionResultsRecalculated, no results if it's not finished

File: models/Events/DeliveryExecutionResultsRecalculated.php [hunk 3]
```diff
@@ -73,7 +73,7 @@ public function isFullyGraded(): bool
         return $this->isFullyGraded;
     }
 
-    public function getTimestamp(): int
+    public function getTimestamp(): ?int
```
Comment (pnal) [inline models/Events/DeliveryExecutionResultsRecalculated.php:76]:
```suggestion
    public function getTimestamp(): int
```

File: models/Events/DeliveryExecutionResultsRecalculated.php (hunk 2 above)
Comment (Karol-Stelmaczonek) [reply models/Events/DeliveryExecutionResultsRecalculated.php:42]: Makes sense, I will remove nulls

**Review by bartlomiejmarszal — APPROVED** [review]

**Review by pnal — APPROVED** [review]

**Pull request merged**